File: icpc/car_washing.py
import sys

def solve():
    data = sys.stdin.buffer.read().strip().split()
    it = iter(data)
    T = int(next(it))
    out_lines = []
    for _ in range(T):
        
        N = int(next(it))
        e1 = int(next(it)); e2 = int(next(it))
        w1 = [0] + [int(next(it)) for _ in range(N)]
        w2 = [0] + [int(next(it)) for _ in range(N)]
        # s1: switch from lane1 -> lane2 after station j (j = 1..N-1)
        s1 = [0] + ([int(next(it)) for _ in range(N-1)] if N > 1 else [])
        # s2: switch from lane2 -> lane1 after station j (j = 1..N-1)
        s2 = [0] + ([int(next(it)) for _ in range(N-1)] if N > 1 else [])
        x1 = int(next(it)); x2 = int(next(it))

        # base (station 1)
        t1 = e1 + w1[1]
        t2 = e2 + w2[1]

        # stations 2..N
        for j in range(2, N+1):
            new_t1 = min(t1 + w1[j], t2 + s2[j-1] + w1[j])
            new_t2 = min(t2 + w2[j], t1 + s1[j-1] + w2[j])
            t1, t2 = new_t1, new_t2

        ans = min(t1 + x1, t2 + x2)
        out_lines.append(str(ans))

    sys.stdout.write("\n".join(out_lines))

if __name__ == "__main__":
    solve()

File: icpc/car_washing.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;
    while (T--) {
        int N;
        cin >> N;

        int e1, e2;
        cin >> e1 >> e2;

        vector<int> w1(N+1), w2(N+1);
        for (int j = 1; j <= N; j++) cin >> w1[j];
        for (int j = 1; j <= N; j++) cin >> w2[j];

        vector<int> s1(N), s2(N); // size N, but only indices [1..N-1] used
        if (N > 1) {
            for (int j = 1; j <= N-1; j++) cin >> s1[j];
            for (int j = 1; j <= N-1; j++) cin >> s2[j];
        }

        int x1, x2;
        cin >> x1 >> x2;

        // Base case (station 1)
        int t1 = e1 + w1[1];
        int t2 = e2 + w2[1];

        // Process stations 2..N
        for (int j = 2; j <= N; j++) {
            int new_t1 = min(t1 + w1[j], t2 + s2[j-1] + w1[j]);
            int new_t2 = min(t2 + w2[j], t1 + s1[j-1] + w2[j]);
            t1 = new_t1;
            t2 = new_t2;
        }

        int ans = min(t1 + x1, t2 + x2);
        cout << ans << "\n";
    }
    return 0;
}
